cmd/performance-profile-creator/cmd/root.go

```diff
@@ -44,22 +44,59 @@ const (
 	infoModeLog  = "log"
 )
 
+const (
+	// defaultLatency refers to the fact that no additional configuration is needed
+	defaultLatency string = "default"
+
+	// lowLatency configures additional set of kernel and tuned arguments for the realtime workload
+	// kernel arguments
+	// 1. nohz_full=${isolated_cores}
+	// 2. tsc=nowatchdog
+	// 3. nosoftlockup
+	// 4. nmi_watchdog=0
+	// 5. mce=off
+	// 6. skew_tick=1
+	//
+	// tuned configuration
+	// 1. stalld enabled
+	// 2. sched_rt_runtime_us=-1
+	// 3. kernel.hung_task_timeout_secs=600
+	// 4. kernel.nmi_watchdog=0
+	// 5. kernel.sched_rt_runtime_us=-1
+	// 6. vm.stat_interval=10
+	lowLatency string = "low-latency"
+
+	// ultraLowLatency in addition to low-latency configuration, disabling CPU P and C states
+	// to guarantee that the CPU will have the lowest responsive time(also meaning high CPU consumption)
+	// kernel Arguments
+	// processor.max_cstate=1
+	// intel_idle.max_cstate=0
+	// intel_pstate=disable
+	// idle=poll
+	// For more information on CPU "C-states" please refer to https://gist.github.com/wmealing/2dd2b543c4d3cff6cab7
+	ultraLowLatency string = "ultra-low-latency"
+)
+
 var (
-	validTMPolicyValues = []string{kubeletconfig.SingleNumaNodeTopologyManagerPolicy, kubeletconfig.BestEffortTopologyManagerPolicy, kubeletconfig.RestrictedTopologyManagerPolicy}
-	validInfoModes      = []string{infoModeLog, infoModeJSON}
+	validTMPolicyValues        = []string{kubeletconfig.SingleNumaNodeTopologyManagerPolicy, kubeletconfig.BestEffortTopologyManagerPolicy, kubeletconfig.RestrictedTopologyManagerPolicy}
+	validInfoModes             = []string{infoModeLog, infoModeJSON}
+	validPowerConsumptionModes = []string{defaultLatency, lowLatency, ultraLowLatency}
 )
 
 // ProfileData collects and stores all the data needed for profile creation
 type ProfileData struct {
-	isolatedCPUs, reservedCPUs string
-	nodeSelector               *metav1.LabelSelector
-	mcpSelector                map[string]string
-	performanceProfileName     string
-	topologyPoilcy             string
-	rtKernel                   bool
-	additionalKernelArgs       []string
-	userLevelNetworking        *bool
-	disableHT                  bool
+	isolatedCPUs              string
+	reservedCPUs              string
+	nodeSelector              *metav1.LabelSelector
+	mcpSelector               map[string]string
+	performanceProfileName    string
+	topologyPolicy            string
+	rtKernel                  bool
+	additionalKernelArgs      []string
+	userLevelNetworking       *bool
+	disableHT                 bool
+	realtimeHint              *bool
+	highPowerConsumptionHints *bool
```

Comment:
s/Hints/Hint   (or is this suppose to be in plural?)

Reply:
done